src/player.rs

```diff
@@ -371,8 +345,11 @@ impl PlayerInternal {
 
                 match packet {
                     Some(Ok(packet)) => {
-                        let buffer = apply_volume(self.state.lock().unwrap().volume,
-                                                  &packet.data);
+                        let buffer = if let Some(ref editor) = stream_editor {
+                            editor.modify_stream(&packet.data)
```

Comment:
This could just be `editor.modify_stream(&mut packet.data)` to modify in place.
(See my comment above)